### PR TITLE
Notify user when credits are too low for full length song playback

### PR DESCRIPTION
**Pull request by peterklingelhofer**
Per [Let the user know they’ve run out of credits](https://community.resonate.is/t/let-the-user-know-theyve-run-out-of-credits/2628), I've implemented a small change here for when `credits < 1`. Open to any suggestions to change the text or improve upon this as well.

Screen Shot 2022-03-16 at 10 02 37 PM

### CODE REVIEW

**Comment by fgblomqvist** [comment]
I like the look/location of the message but not too sure about the phrasing. Mainly since less than one credit can still go quite far. What's interesting here is that you might not be able to afford a play on one song, but then you can on another (depending on how many listens you've had). Eh maybe this message is good enough as a catch-all. Could always be changed later if someone comes up with something better 🙂

**Review by fgblomqvist — APPROVED** [review]

File: packages/player-component/index.js [hunk 1]
```diff
@@ -326,12 +326,15 @@ class Player extends Nanocomponent {
...
 
     return html`
       <div class="player-component flex flex-column h-100">
         ${!isAuthenticated
-          ? html`<a class="flex flex-row w-100 justify-center bb b--light-silver no-underline" href="/login" target="_blank" rel="noopener noreferer">Log in to listen to full song</a>`
-          : ''}
+          ? html`<a class=${infoBarClasses} href="/login" target="_blank" rel="noopener noreferer">Log in to listen to full song</a>`
+          : Number(this.state.user.credits) < 1
+            ? html`<a class=${infoBarClasses}>You have less than one credit remaining</a>`
+            : ''}
```
Comment (fgblomqvist) [inline packages/player-component/index.js]: As a side note: nested ternaries tend to be frowned upon due to the significantly increased reading complexity.  Might be nice to just figure this out outside of the return statement using some regular variables/if statements and then just render here. Just a thought though.

Comment (peterklingelhofer) [reply packages/player-component/index.js]: Good call. Addressed here: https://github.com/resonatecoop/stream/pull/213/commits/3bbf330b7d87897a0a4a2185c21125e89a612adf

**Comment by peterklingelhofer** [comment]
> I like the look/location of the message but not too sure about the phrasing. Mainly since less than one credit can still go quite far. What's interesting here is that you might not be able to afford a play on one song, but then you can on another (depending on how many listens you've had). Eh maybe this message is good enough as a catch-all. Could always be changed later if someone comes up with something better 🙂

Totally agreed, I'd thought this too. Not great but this seemed like the simplest solution. Very open to changing it if anyone can think of better phrasing or more appropriate number comparison.

**Comment by peterklingelhofer** [comment]
> I wonder if it’s possible to only show the message when you don’t have enough credits for the current track. Reusing whichever logic figures out that the track play should be limited to 45 seconds? And the message might say e.g. ‘You don’t have enough credits to play the current track in full’.

Going to try to iterate on this current approach per the suggestion from the forum thread, and see if we can improve upon the current solution.

**Comment by peterklingelhofer** [comment]
It seems that to check if we're in `preview` mode or not, [this line](https://github.com/resonatecoop/stream/blob/5e4e0418e5ba39d0fe21efd88e33cd2f29c27865/packages/player-component/index.js#L81) and a few others like it check to see if `this.state.user.credits < 0.002`, so I've implemented that here. I was surprised to see this, since I thought the final play on a track should cost something like 90 cents. Given that this is our current implementation, I think this solution should work for now.

**Comment by fgblomqvist** [comment]
This check is actually a real problem, and a quite serious one at that imo: https://github.com/resonatecoop/stream/issues/217

**Comment by fgblomqvist** [comment]
Ideally there should be a global function of some sort that can return whether you can afford to listen to the full version of the song atm.

**Comment by fgblomqvist** [comment]
I think this could be merged for now since, like you mentioned previously, it at least uses the same mechanic (even if it's faulty) as the preview mode. It's still better than nothing for someone with 0 credits.

**Review by fgblomqvist — APPROVED** [review]

**Comment by fgblomqvist** [comment]
The previously mentioned issue has been solved. I'm not sure how, but might be worth taking a second look at this to see if preview mode detection has changed (and how).

**Pull request merged**